### DISCUSSION TITLE
Duplicate decoders?

### DISCUSSION THREAD

**Discussion by habub** [merbanan/rtl_433#2217] [General]
Hello,
So I'm testing 2 new decoders. They're identical in modulation (ook_pwm) and timing, but the payload is different.
Each decoder has a different fle name (and r_device .name), of course. I added both devices to the cmake list, and corresponding obj files are created (_cmake -G "Visual Studio 17 2022" ._.).

But, when I build the program (_cmake --build ._) only 1 of the decoders is actually used. The other is discarded and doesn't appear in the supported decoders list (-R).

What am I missing?

**Comment by zuckschwerdt** [comment]
File name and timings don't matter, the important thing is a unique `r_device` symbol, i.e. `r_device unique_here = `
Then add that symbol in `include/rtl_433_devices.h` as `DECL(unique_here) \`

**Comment by habub** [comment]
Hi,
When I build the program I only get the Debug folder with the executable in it.
Should I also expect a Release folder and binary? Is there a difference?

**Reply by zuckschwerdt** [reply]
Somebody working in Windows might be able to better help. As far as I have tested you don't need to ru cmake, it's all integrated in Visual Studio, and just build there. You can switch the profile to Release if you like. But Debug is also fine, the file is just bigger then.
